src/mapping/RadialBasisFctSolver.h

```diff
@@ -0,0 +1,179 @@
+  _qr = buildMatrixCLU(basisFunction, inputMesh, activeAxis).colPivHouseholderQr();
+
+  PRECICE_CHECK(_qr.isInvertible(),
+                "The interpolation matrix of the RBF mapping from mesh {} to mesh {} is not invertable. "
+                "This means that the mapping problem is not well-posed. "
+                "Please check if your coupling meshes are correct. Maybe you need to fix axis-aligned mapping setups "
+                "by marking perpendicular axes as dead?",
+                "this->input()->getName()", " this->output()->getName()");
+
+  // Second, assemble evaluation matrix
+  _matrixA = buildMatrixA(basisFunction, inputMesh, outputMesh, activeAxis);
+}
+
+Eigen::VectorXd RadialBasisFctSolver::solveConservative(const Eigen::VectorXd &inputData) const
+{
+  // TODO: Avoid temporary allocations
+  PRECICE_ASSERT(inputData.size() == _matrixA.rows());
+  Eigen::VectorXd Au = _matrixA.transpose() * inputData;
+  PRECICE_ASSERT(Au.size() == _matrixA.cols());
+  return _qr.solve(Au);
+}
+
+Eigen::VectorXd RadialBasisFctSolver::solveConsistent(const Eigen::VectorXd &inputData) const
+{
+  PRECICE_ASSERT(inputData.size() == _matrixA.cols());
+  Eigen::VectorXd p = _qr.solve(inputData);
+  PRECICE_ASSERT(p.size() == _matrixA.cols());
+  return _matrixA * p;
+}
+
+void RadialBasisFctSolver::clear()
+{
+  _matrixA = Eigen::MatrixXd();
+  _qr      = Eigen::ColPivHouseholderQR<Eigen::MatrixXd>();
+}
+
+// ------- Non-Member Functions ---------
+
+/// Deletes all dead directions from fullVector and returns a vector of reduced dimensionality.
+static inline double computeSquaredDifference(
+    const std::array<double, 3> &u,
+    std::array<double, 3>        v,
+    const std::array<bool, 3> &  activeAxis)
+{
+  // Substract the values and multiply out dead dimensions
+  for (unsigned int d = 0; d < v.size(); ++d) {
+    v[d] = (u[d] - v[d]) * static_cast<int>(activeAxis[d]);
+  }
+  return std::accumulate(v.begin(), v.end(), static_cast<double>(0.), [](auto &res, auto &val) { return res + val * val; });
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+static Eigen::MatrixXd buildMatrixCLU(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, std::array<bool, 3> activeAxis)
+{
+  const unsigned int inputSize      = inputMesh.vertices().size();
+  const unsigned int deadDimensions = std::count(activeAxis.begin(), activeAxis.end(), false);
+  // Treat the 2D case as 3D case with dead axis
+  const unsigned int dimensions = 3;
+  const unsigned int polyparams = 1 + dimensions - deadDimensions;
+  const unsigned int n          = inputSize + polyparams; // Add linear polynom degrees
+
+  PRECICE_ASSERT((inputMesh.getDimensions() == 3) || activeAxis[2] == false);
+  PRECICE_ASSERT(inputSize >= 1 + polyparams, inputSize);
+
+  Eigen::MatrixXd matrixCLU(n, n);
+  matrixCLU.setZero();
+
+  for (unsigned int i = 0; i < inputSize; ++i) {
+    for (unsigned int j = i; j < inputSize; ++j) {
```

Comment:
We could introduce some iteration strategy and pass both meshes and the active axis, but this will be most certainly more difficult to read and more code than those 2 short loops.

I strongly tend to leave it as is.